sensing/autoware_pointcloud_preprocessor/src/distortion_corrector/distortion_corrector_node.cpp

```diff
@@ -92,10 +93,32 @@ void DistortionCorrectorComponent::onPointCloud(PointCloud2::UniquePtr pointclou
     return;
   }
 
-  distortion_corrector_->setPointCloudTransform(base_frame_, pointcloud_msg->header.frame_id);
-
+  distortion_corrector_->set_pointcloud_transform(base_frame_, pointcloud_msg->header.frame_id);
   distortion_corrector_->initialize();
-  distortion_corrector_->undistortPointCloud(use_imu_, *pointcloud_msg);
+
+  if (update_azimuth_and_distance_ && !angle_conversion_opt_.has_value()) {
+    angle_conversion_opt_ = distortion_corrector_->try_compute_angle_conversion(*pointcloud_msg);
+    if (angle_conversion_opt_.has_value()) {
+      RCLCPP_INFO(
+        this->get_logger(),
+        "Success to get the conversion formula between cartesian coordinates and LiDAR azimuth "
+        "coordinates");
+    } else {
+      RCLCPP_WARN_STREAM_THROTTLE(
+        this->get_logger(), *this->get_clock(), 10000 /* ms */,
+        "Failed to get the angle conversion between Cartesian coordinates and LiDAR azimuth "
+        "coordinates. This pointcloud will not update azimuth and distance");
+      angle_conversion_failure_num_++;
+
+      if (angle_conversion_failure_num_ > failure_tolerance) {
+        throw std::runtime_error(
+          "Angle conversion failed more than " + std::to_string(failure_tolerance) +
```

Comment:
While I get the feeling, I do not think throwing here is good for our projects (imagine if we throw in the middle of an experiment/production).
In my opinion, we could send an error and fallback to the default case while being verbose about this fact.

Also, I do not believe the use of a tolerance is warranted. For this cases, it works or not, right?

Reply:
Understand your consideration, before we think it is better for the user to set the parameter to False after 2 seconds (if it is failed).
Remove the runtime error and set the logging from warning to error in https://github.com/autowarefoundation/autoware.universe/pull/8380/commits/4bc0df1ea17e80b390bca04ba2f66fd0df39afdb